service/scaladsl/server/src/main/scala/com/lightbend/lagom/scaladsl/server/LagomApplicationLoader.scala

```diff
@@ -213,8 +213,11 @@ abstract class LagomApplication(context: LagomApplicationContext)
 
   @deprecated(message = "prefer `config` using typesafe Config instead", since = "1.4.0")
   override lazy val configuration: Configuration = {
-    val additionalConfig = new Configuration(additionalConfiguration.configuration)
-    Configuration.load(environment) ++ context.playContext.initialConfiguration ++ additionalConfig
+    val initialConfiguration = Configuration.load(environment) ++ context.playContext.initialConfiguration
+    val additionalConfig = new Configuration(
+      additionalConfiguration(initialConfiguration).configuration.withFallback(additionalConfiguration.configuration)
```

Comment:
This should be carefully documented and tested.

Actually, could this code somehow be moved into `ProvidesAdditionalConfiguration` (maybe as a private method?) to keep the logic more located?

Reply:
that seems like a good idea. i will update the PR